### PR TITLE
added app with db and 3 models: Car, Driver, Manufacturer

### PR DESCRIPTION
**Pull request by vicire**
focused on Manufacturer

### CODE REVIEW

**Review by Max-Pochepets — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (Sofasmile) [inline .travis.yml]: let's add empty line at the end of the file

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,61 @@
...
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (Sofasmile) [inline pom.xml]: do we need this dependency here? let's remove it

File: src/main/java/core/basesyntax/db/Storage.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Manufacturer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static final List<Manufacturer> manufacturerStorage = new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/db/Storage.java]:
```suggestion
    public static final List<Manufacturer> manufacturers = new ArrayList<>();
```

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturerStorage;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldManufacturer = get(manufacturer.getId()).get();
+        oldManufacturer.setCountry(manufacturer.getCountry());
+        oldManufacturer.setName(manufacturer.getName());
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        return oldManufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        if (get(id).isPresent()) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: read about method `removeIf`

File: src/main/java/core/basesyntax/model/Car.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer) {
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = new ArrayList<>();
+    }
+
+    public long getId() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/model/Car.java]: why do we use primitive here?

File: src/main/java/core/basesyntax/model/Manufacturer.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
+package core.basesyntax.model;
+
+import java.util.Objects;
+
+public class Manufacturer {
+    private long id;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/model/Manufacturer.java:6]: how about wrapper?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (vicire) [reply .travis.yml]: Fixed

File: pom.xml (hunk 2 above)
Comment (vicire) [reply pom.xml]: Fixed

File: src/main/java/core/basesyntax/db/Storage.java (hunk 3 above)
Comment (vicire) [reply src/main/java/core/basesyntax/db/Storage.java]: Fixed

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 4 above)
Comment (vicire) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: Fixed

File: src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java (hunk 5 above)
Comment (vicire) [reply src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java]: Fixed

File: src/main/java/core/basesyntax/model/Car.java (hunk 6 above)
Comment (vicire) [reply src/main/java/core/basesyntax/model/Car.java]: Fixed

File: src/main/java/core/basesyntax/model/Manufacturer.java (hunk 7 above)
Comment (vicire) [reply src/main/java/core/basesyntax/model/Manufacturer.java:6]: Fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**